core/src/main/scala/no/nrk/bigquery/BQPartitionType.scala

```diff
@@ -21,13 +21,24 @@ import java.time.{LocalDate, YearMonth}
   *
   * It is meant that when we add support for more partition schemes, the first step is to add a new subtype here.
   */
-sealed trait BQPartitionType[+Param]
+sealed trait BQPartitionType[+Param] {
+  def partitionField: Option[Ident] = {
+    val asType = this.asInstanceOf[BQPartitionType[Any]]
+    asType match {
+      case BQPartitionType.DatePartitioned(field) => Some(field)
+      case BQPartitionType.MonthPartitioned(field) => Some(field)
+      case BQPartitionType.RangePartitioned(field, _) => Some(field)
+      case _: BQPartitionType.Sharded => None
+      case _: BQPartitionType.NotPartitioned => None
+    }
+  }
+}
 
 object BQPartitionType {
 
   final case class DatePartitioned(field: Ident) extends BQPartitionType[LocalDate]
-
   final case class MonthPartitioned(field: Ident) extends BQPartitionType[YearMonth]
+  final case class RangePartitioned(field: Ident, range: BQRange = BQRange.default) extends BQPartitionType[Long]
```

Comment:
BigQuery refer to this a integer range partition. Should we do the same?


https://cloud.google.com/bigquery/docs/partitioned-tables#integer_range